src/interpreter_funcs/boolean_difference.rs

```diff
@@ -0,0 +1,109 @@
+use std::error;
+use std::f32;
+use std::fmt;
+use std::sync::Arc;
+
+use nalgebra::Vector3;
+
+use crate::interpreter::{
+    Float3ParamRefinement, Func, FuncError, FuncFlags, FuncInfo, ParamInfo, ParamRefinement, Ty,
+    UintParamRefinement, Value,
+};
+use crate::mesh::voxel_cloud::VoxelCloud;
+
+#[derive(Debug, PartialEq)]
+pub enum FuncBooleanDifferenceError {
+    WeldFailed,
```

Comment:
This is missing the error case when the difference produces a empty voxel cloud.